src/Mod/Path/PathScripts/PathProbe.py

```diff
@@ -0,0 +1,107 @@
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+from __future__ import print_function
+
+import FreeCAD
+import Path
+import PathScripts.PathLog as PathLog
+import PathScripts.PathOp as PathOp
+import PathScripts.PathUtils as PathUtils
+
+from PySide import QtCore
+
+__title__ = "Path Probing Operation"
+__author__ = "sliptonic (Brad Collette)"
+__url__ = "http://www.freecadweb.org"
+__doc__ = "Path Probing operation."
+
+if False:
+    PathLog.setLevel(PathLog.Level.DEBUG, PathLog.thisModule())
+    PathLog.trackModule(PathLog.thisModule())
+else:
+    PathLog.setLevel(PathLog.Level.INFO, PathLog.thisModule())
+
+
+# Qt tanslation handling
+def translate(context, text, disambig=None):
+    return QtCore.QCoreApplication.translate(context, text, disambig)
+
+
+class ObjectProbing(PathOp.ObjectOp):
+    '''Proxy object for Probing operation.'''
+
+    def opFeatures(self, obj):
+        '''opFeatures(obj) ... Probing works on the stock object.'''
+        return PathOp.FeatureDepths | PathOp.FeatureHeights | PathOp.FeatureTool
+
+    def initOperation(self, obj):
+        obj.addProperty("App::PropertyLength", "Xoffset", "Probe", QtCore.QT_TRANSLATE_NOOP("App::Property", "X offset between tool and probe"))
+        obj.addProperty("App::PropertyLength", "Yoffset", "Probe", QtCore.QT_TRANSLATE_NOOP("App::Property", "Y offset between tool and probe"))
+        obj.addProperty("App::PropertyInteger", "PointCountX", "Probe", QtCore.QT_TRANSLATE_NOOP("App::Property", "Number of points to probe in X direction"))
+        obj.addProperty("App::PropertyInteger", "PointCountY", "Probe", QtCore.QT_TRANSLATE_NOOP("App::Property", "Number of points to probe in Y direction"))
+        obj.addProperty("App::PropertyFile", "OutputFileName", "Path", QtCore.QT_TRANSLATE_NOOP("App::Property", "The output location for the probe data to be written"))
+
+    def nextpoint(self, startpoint=0.0, endpoint=0.0, count=3):
+        curstep = 0
+        dist = (endpoint - startpoint) / (count - 1)
+        while curstep <= count-1:
+            yield startpoint + (curstep * dist)
+            curstep += 1
+
+    def opExecute(self, obj):
+        '''opExecute(obj) ... generate probe locations.'''
+        PathLog.track()
+        self.commandlist.append(Path.Command("(Begin Probing)"))
+
+        stock = PathUtils.findParentJob(obj).Stock
+        bb = stock.Shape.BoundBox
+
+        openstring = '(PROBEOPEN {})'.format(obj.OutputFileName)
```

Comment:
Isn't this specific to LinuxCNC?
Shouldn't cause any harm though. If this becomes part of the official FC post processor api we should probably document it.

Reply:
Unfortunately yes. It's very linuxcnc specific. I don't think there's much of a standard for this kind of round-trip interaction. We may have to revise this if one emerges. Maybe the machinekit workbench will alleviate the need. <hint hint> :-)